### ISSUE TITLE
Issue running exported edgetpu model in Coral USB

### ISSUE THREAD

**Issue by bhakav** [ultralytics/ultralytics#3669]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Env - Ubuntu 18.04LTS
Yolov8 - XLNS

I am successfully able to export existing YOLOV8 model to edgetpu

*******************************************************
#convert to edgeTpu
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n.pt')  # load an official model
model.export(format='edgetpu')
*******************************************************
Output
Ultralytics YOLOv8.0.124 🚀 Python-3.8.5 torch-2.0.1+cu117 CPU
YOLOv8n summary (fused): 168 layers, 3151904 parameters, 0 gradients, 8.7 GFLOPs

PyTorch: starting from yolov8n.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 84, 8400) (6.2 MB)
requirements: Ultralytics requirements "onnx2tf>=1.7.7" "sng4onnx>=1.0.1" "onnxsim>=0.4.17" "onnx_graphsurgeon>=0.3.26" "tflite_support" not found, attempting AutoUpdate...
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com/
Collecting onnx2tf>=1.7.7
  Downloading onnx2tf-1.14.4-py3-none-any.whl (371 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 372.0/372.0 kB 1.1 MB/s eta 0:00:00
Collecting sng4onnx>=1.0.1
  Downloading sng4onnx-1.0.1-py3-none-any.whl (5.8 kB)
Collecting onnxsim>=0.4.17
  Downloading onnxsim-0.4.33-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.2/2.2 MB 1.1 MB/s eta 0:00:00
Collecting onnx_graphsurgeon>=0.3.26
  Downloading https://developer.download.nvidia.com/compute/redist/onnx-graphsurgeon/onnx_graphsurgeon-0.3.27-py2.py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.1/42.1 kB 9.3 MB/s eta 0:00:00
Collecting tflite_support
  Downloading tflite_support-0.4.3-cp38-cp38-manylinux2014_x86_64.whl (60.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.8/60.8 MB 1.5 MB/s eta 0:00:00
Requirement already satisfied: onnx in /home/spot/anaconda3/lib/python3.8/site-packages (from onnxsim>=0.4.17) (1.13.1)
Requirement already satisfied: rich in /home/spot/anaconda3/lib/python3.8/site-packages (from onnxsim>=0.4.17) (12.5.1)
Requirement already satisfied: numpy in /home/spot/anaconda3/lib/python3.8/site-packages (from onnx_graphsurgeon>=0.3.26) (1.24.2)
Requirement already satisfied: absl-py>=0.7.0 in /home/spot/anaconda3/lib/python3.8/site-packages (from tflite_support) (1.4.0)
Requirement already satisfied: protobuf<4,>=3.18.0 in /home/spot/anaconda3/lib/python3.8/site-packages (from tflite_support) (3.20.3)
Collecting pybind11>=2.6.0
  Downloading pybind11-2.10.4-py3-none-any.whl (222 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 222.3/222.3 kB 650.7 kB/s eta 0:00:00
Requirement already satisfied: flatbuffers>=2.0 in /home/spot/anaconda3/lib/python3.8/site-packages (from tflite_support) (23.5.9)
Collecting sounddevice>=0.4.4
  Downloading sounddevice-0.4.6-py3-none-any.whl (31 kB)
Requirement already satisfied: CFFI>=1.0 in /home/spot/anaconda3/lib/python3.8/site-packages (from sounddevice>=0.4.4->tflite_support) (1.15.1)
Requirement already satisfied: typing-extensions>=3.6.2.1 in /home/spot/anaconda3/lib/python3.8/site-packages (from onnx->onnxsim>=0.4.17) (4.4.0)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /home/spot/anaconda3/lib/python3.8/site-packages (from rich->onnxsim>=0.4.17) (2.15.1)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /home/spot/anaconda3/lib/python3.8/site-packages (from rich->onnxsim>=0.4.17) (0.9.1)
Requirement already satisfied: pycparser in /home/spot/anaconda3/lib/python3.8/site-packages (from CFFI>=1.0->sounddevice>=0.4.4->tflite_support) (2.21)
Installing collected packages: sng4onnx, pybind11, onnx2tf, sounddevice, onnxsim, onnx_graphsurgeon, tflite_support
Successfully installed onnx2tf-1.14.4 onnx_graphsurgeon-0.3.27 onnxsim-0.4.33 pybind11-2.10.4 sng4onnx-1.0.1 sounddevice-0.4.6 tflite_support-0.4.3

requirements: 5 packages updated per ('onnx', 'onnx2tf>=1.7.7', 'sng4onnx>=1.0.1', 'onnxsim>=0.4.17', 'onnx_graphsurgeon>=0.3.26', 'tflite_support', 'onnxruntime')
requirements: ⚠️ Restart runtime or rerun command for updates to take effect

TensorFlow SavedModel: starting export with tensorflow 2.11.1...

ONNX: starting export with onnx 1.13.1 opset 17...
ONNX: simplifying with onnxsim 0.4.33...
============= Diagnostic Run torch.onnx.export version 2.0.1+cu117 =============
verbose: False, log level: Level.ERROR
======================= 0 NONE 0 NOTE 0 WARNING 0 ERROR ========================

ONNX: export success ✅ 3.8s, saved as yolov8n.onnx (12.2 MB)

TensorFlow SavedModel: running 'onnx2tf -i yolov8n.onnx -o yolov8n_saved_model -nuo --non_verbose -oiqt -qt per-tensor'

****************************************

When I try to run the model on Coral edge device with support for edgeTPU , I get below error when loading the model

"
Loading ../all_models/yolov8n_full_integer_quant.tflite with ../all_models/yolo.txt labels.
Segmentation fault (core dumped)
"

I tried different built in edgeTPU models and they all work great.

Any advice will be  helpful.

V/R,
Bhakav

**Comment by glenn-jocher** [comment]
That's fantastic to hear @shashichilappagari! Your team at DeGirum has done an excellent job quantizing and porting Ultralytics object detection models, including YOLOv8, onto the Edge TPU. It's very exciting to hear about the variety of specialized models you've created for tasks such as face, hand, car, and license plate detection.

Making these models available on your cloud platform will be a great help to many users. Your platform's option to try each model on different configurations like OpenVINO CPU (float), TFLite CPU (float), and TFLite EdgeTPU (quant) is commendable.

We appreciate you sharing the availability of a Google colab notebook for running the models. We are sure users will find it useful. As always, we encourage our users to reach out and open an issue for any questions or support needed.

Your live demonstration sounds intriguing! We hope the community benefits from your initiative and learns more about the capabilities of YOLOv8 models.

Keep up the great work!

**Comment by bhakav** [comment]
> @bhakav Our team at DeGirum has successfully quantized and ported Ultralytics object detection models to Edge TPU. Specifically: YOLOv5nu, YOLOV5su, YOLOv8n, and YOLOv8s, with SiLU or ReLU6 activation functions. Each of these has been trained for a full COCO object detection task, as well as specialized models for face, hand, car, and license plate detection!
> 
> We are making these models available on our cloud platform at: https://degirum.com/#sign-up
> 
> On the platform, you can try each model on: OpenVINO CPU (float), TFLite CPU (float), and TFLite EdgeTPU (quant), for a total of 120 possible combinations!
> 
> Once signed up, you can run the models in the browser or in this Google colab notebook: https://colab.research.google.com/github/DeGirum/PySDKExamples/blob/main/UltralyticsObjectDetectionImage.ipynb
> 
> For any questions or support, please open an issue on: https://github.com/DeGirum/PySDKExamples
> 
> If anyone’s interested, we will be doing a live demonstration of using these new models at: https://www.linkedin.com/events/theartofyolo-part1-unleashingth7092593180545540096/

This is great news. The issue I am having is when running my code after converting  yolov8n model to TFLite EdgeTPU . Do you have any solution for that ?  Can I download your model to test them or we can only try using your cloud platform ?

Pls advice

**Comment by shashichilappagari** [comment]
@bhakav The cloud platform is a quick way for you to get started to make sure that the models are working for the HW option you need. You can install our PySDK locally and run the models on your hardware. See https://docs.degirum.com/content/ for details. The models are downloaded automatically when you run them locally. You can download the models from the portal and run them without PySDK but you need to write your own pre-processing and post-processing code. For the exported models to have good speed and accuracy, we had to make some changes to how the models are exported. Specifically, the code related to decoding of bounding boxes is excluded from the model and pushed to the post-processing. Please reach out to us if you still have difficulties in running the models. Our team can help you get setup.

**Comment by glenn-jocher** [comment]
@bhakav, the cloud platform offered by DeGirum is intended to help you get started quickly and validate the models on your hardware configuration. By installing their PySDK locally, the models will be automatically downloaded when you run them on your hardware. 

You do have the option to download the models directly from the portal and run them without PySDK. However, please be aware that you will need to write your own code for pre-processing and post-processing.

Please note that to optimize both speed and accuracy, certain changes were made during the model exporting process. For instance, the code related to decoding of bounding boxes is not included in the model and is handled during post-processing.

If you continue to face difficulties running the models, please don't hesitate to reach out. The team at DeGirum is equipped to provide assistance in getting you set up.

**Comment by circa10a** [comment]
@shashichilappagari can you confirm if the ported model available from your cloud platform will work with yolov8 + local edge tpu if downloaded?

**Comment by shashichilappagari** [comment]
@circa10a yes, the downloaded model will work locally with our pysdk, provided you already have edge tpu driver installed. Please let me know if you encounter any issues.

**Comment by circa10a** [comment]
> @circa10a yes, the downloaded model will work locally with our pysdk, provided you already have edge tpu driver installed. Please let me know if you encounter any issues.

But will the model work with this ultralytics library? (Not the aforementioned pysdk)

**Comment by shashichilappagari** [comment]
> > @circa10a yes, the downloaded model will work locally with our pysdk, provided you already have edge tpu driver installed. Please let me know if you encounter any issues.
> 
> But will the model work with this ultralytics library? (Not the aforementioned pysdk)

No, it will not work with ultralytics library.

**Comment by shashichilappagari** [comment]
@circa10a We made a thin wrapper to run exported models in onnxruntime, tflite and on edge tpu: https://github.com/DeGirum/yolo_inference
The repo also has some sample models for experimentation.
